state-chain/chains/src/sol/sol_tx_core/address_derivation.rs

```diff
@@ -0,0 +1,134 @@
+//! Provides a interface to deriving addresses for the purposes of DepositChannels
+//! as well as deriving ATA for token operations (such as fetch or transfer).
+
+use cf_primitives::ChannelId;
+use sol_prim::AccountBump;
+
+use crate::sol::{consts::*, AddressDerivationError, DerivedAddressBuilder, SolAddress};
+use core::str::FromStr;
+
+/// Derive address for a given channel ID
+pub fn derive_deposit_address(
+	channel_id: ChannelId,
+	vault_program: SolAddress,
+) -> Result<(SolAddress, AccountBump), AddressDerivationError> {
```

Comment:
Passing a comment from Dan:
"We could create a DerivedAddress struct instead of using a raw tuple."

Reply:
THis is now addressed.